### ISSUE TITLE
Export linked .rvt in a single IFC file

### ISSUE THREAD

**Issue by jlaiph** [Autodesk/revit-ifc#439]
I know this is not a new topic to export a single Ifc file from multiple linked revit files. Howvever, it will be great if this feature can this be achieved in Revit rather than go through Navisworks and iConstruct or Codemill IFC Exporter to export the appended files into one IFC model. The current binding process in Revit for the linked .rvt file still result in loss of geometry information.

**Comment by jlaiph** [comment]
Thanks.

**Comment by Moult** [comment]
Why is this closed? The functionality still does not exist so the problem is still outstanding so this should be opened.

**Comment by jlaiph** [comment]
Hi, the reason is this will be a redundant proposal.  Anyway, quick solution for me now is to merge multiple IFC files using a third-party software.

**Comment by Moult** [comment]
It's not a redundant proposal - the result still cannot be achieved.

**Comment by AngelVelezSosa** [comment]
I assume he meant that it was redundant to the Revit public roadmap, but I have no issues with it also being open here.
I do have one question as I look into this:
What is the expected tree structure for a federated IFC file?  Is it one IFCBUILDING per link, all under the same IFCSITE?   Is there one IFCSITE/IFCBUILDING per file?  Do they all share the same IFCBUILDING/IFCSITE?  I would think it would really have to be one IFCSITE/IFCBUILDING per file (meaning, the only common thing is the IFCPROJECT), but I'm curious what others think.

**Comment by jlaiph** [comment]
That is an interesting question. I would like to know as well. For me, it should be user defined. The "3rd party" software I am using now is able to define "Same IFCSite", "Same IFCBuilding" or "Same IFCFloor" in such hierarchy. The reason is I have received IFC files from external, most of the time they only have IFCProject and IFCSite with wrong coordination exported. Most of the time I cannot simply link these IFC files directly in Revit and need to re-harmonize the information with "3rd party" software.

**Comment by BIMLUND** [comment]
I agree @AngelVelezSosa. I would say same IFCBuilding under the same IFCsite. But ofcource it would be nice to choose an option when exporting.

**Comment by AngelVelezSosa** [comment]
An option is possible.  Using the same site or building does mean that we would then only use the Psets for the site or building from the "main" file and ignore values from the linked files.  I think what makes sense is something like the following:

[ ] Include Linked Files: As separate IFC, As separate sites, As separate buildings, As same building (I'll need to think about the wording).

**Comment by Moult** [comment]
One very common usecase is where the disciplines divide the model into multiple Revit files: architectural, mech, elec, hyd, fire, etc. The arch model has Revit rooms (IFC spaces) but the others do not (or shouldn't, or have it linked or copy monitored / etc). I know it's adding more complexity, but if there was a solution to also somehow allow the services models to be merged (i.e IFC contained in relationships) with the rooms of the arch model in a single IFC, that would be great.

For example COBie requires this, so does the upcoming IFC BEM requirement.

**Comment by AngelVelezSosa** [comment]
What do you mean by "merged"?  Do you mean that the MEP models would be contained in the spatial hierarchy of the architectural spaces?

**Comment by Moult** [comment]
Indeed. Cobie and IFC BEM both require all maintainable assets (mostly services) to be contained in spaces that they are to be maintained from (typically found in the arch model).

Current workarounds are either to copy paste rooms (which we tried unsuccessfully) or to manually enter in and patch afterwards (which is what we now instruct all services to do).

**Comment by AngelVelezSosa** [comment]
I'll have to see if Revit itself supports placing a linked object in a host space.
